stores/sql/main.go

```diff
@@ -244,6 +244,49 @@ func ContractRoots(ctx context.Context, tx sql.Tx, fcid types.FileContractID) ([
 	return roots, nil
 }
 
+// ContractRootsDiff returns the indices, offset by the given offset, of roots
+// that are not in the contract.
+func ContractRootsDiff(ctx context.Context, tx sql.Tx, fcid types.FileContractID, roots []types.Hash256, offset uint64) (indices []uint64, err error) {
```

Comment:
I've put this up for review but I'm looking into this a little bit more because we can probably avoid the temporary table here.